Added umbrella header to allow for including the framework in playgrounds

CalQL8ed-K-OS opened the pull request:
see these links for some details
https://forums.swift.org/t/recent-swiftpm-features-and-enhancements/13807
https://forums.swift.org/t/cant-experiment-swiftnio-in-playground/15020/6

Commit:
Added umbrella header to allow for including the framework in playgrounds

see these links for some details
https://forums.swift.org/t/recent-swiftpm-features-and-enhancements/13807
https://forums.swift.org/t/cant-experiment-swiftnio-in-playground/15020/6

Commit:
rename from upper to lower case to see if it fixes import problem in playground

Commit:
updated .swift-version

Commit:
Revert "rename from upper to lower case to see if it fixes import problem in playground"

This reverts commit 2dbf9bfd165b24ca846bb513b098973a0b1c4736.

Commit:
remove module.modulemap as per linked discussion

https://forums.swift.org/t/generated-xcodeproj-for-c-language-target/15192/2

ffried commented on Zip/minizip/include/Minizip.h:
```diff
@@ -0,0 +1,15 @@
+
+#import "unzip.h"
+#import "zip.h"
+
```
An `#import "crypt.h"` is missing here.
Compiling also warns about it:
```
<module-includes>:1:1: warning: umbrella header for module 'Minizip' does not include header 'crypt.h'
#include "/Zip/Zip/minizip/include/Minizip.h"
^
```

CalQL8ed-K-OS replied:
Hey Florian, I addressed that warning, but in the process came across a few more things worth addressing.
Most notably, I overwrote the project with the results of `swift package generate-xcodeproj` because the project wasn't building via Xcode due to the removal of the `module.modulemap` file.
This may endanger the desirability of the original SPM support PR back into the root **marmelroy/Zip** repo, so you may want to reconsider my PR.
I've made the changes in a series of commits so you may want to cherry-pick whichever bits you like.
¡Thanks for the feedback!

ffried replied:
@CalQL8ed-K-OS Yea, overwriting the Xcode project isn't a good idea for non-SPM projects. 🙂 
I'll check if I can manually integrate your changes. Maybe I can just tell SPM to ignore the module map.

ffried reviewed (changes requested):
Generally looks good. Only small warning to fix. 🙂

xavi-matos committed:
Address SPM warning about missing include

Message from `swift build` was:
```
<module-includes>:1:1: warning: umbrella header for module 'Minizip' does not include header 'crypt.h'
#import "/Users/CalQL8ed-K-OS/Workbench/open-source/Zip/Zip/minizip/include/Minizip.h"
```

xavi-matos committed:
Address warning in Xcode about precision loss

xavi-matos committed:
Overwrote project with results of `swift package generate-xcodeproj`

The project was not building due to the removal of the file `module.modulemap` which collected the C code into the `import`-able module *Minizip*.

xavi-matos committed:
Updated project through recommended changes feature of Xcode

ffried commented:
@CalQL8ed-K-OS So, I've finally found time to look at your changes again and integrate them in my branch.
Like I mentioned, we can't re-generate the Xcode Project with SPM, since this would break non-SPM projects. But I've nevertheless managed to get rid of (or actually move) the module.modulemap file.

Could you maybe check if this now works with playgrounds?

I'll close this PR since its main feature should be integrated in the branch now.

ghost commented:
> @CalQL8ed-K-OS So, I've finally found time to look at your changes again and integrate them in my branch.
> Like I mentioned, we can't re-generate the Xcode Project with SPM, since this would break non-SPM projects. But I've nevertheless managed to get rid of (or actually move) the module.modulemap file.
> 
> Could you maybe check if this now works with playgrounds?
> 
> I'll close this PR since its main feature should be integrated in the branch now.

I've created a new PR, #2, which adds a playground and demonstrates that this branch does indeed work with playgrounds now.
FYI This playground was created using [John Sundell's Playground CLI Tool](https://github.com/JohnSundell/Playground), specifically the command was `playground -t Zip.playground -d ./Zip.xcodeproj`
